src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs

```diff
@@ -80,7 +80,7 @@ public string ToExpression()
     /// <summary>
     /// Indicates that a FormulaValue should be copied before being mutated, for Copy on Write semantics.
     /// </summary>
-    internal interface IMutationCopy
+    public interface IMutationCopy

```

Comment:
>public [](http://example.com/codeflow?start=4&length=6)

this is one thing to consider...

Should we keep this internal. and instead expose as a virtual on RecordValue?

Reply:
I think this should be public to allow our external customers/users to implement this if needed